File: src/cs/Phyllochron.cs
public class Phyllochron
{
    public double phyllochron;
    public double pastMaxAI;
    public Phyllochron(double _phyllochron,double _pastMaxAI)
    {
        phyllochron=_phyllochron;
        pastMaxAI=_pastMaxAI;
    }
}

public static class Estimation_Phyllochron
{
    public static Phyllochron CalculatePhyllochron(double fixPhyll,double leafNumber,double lincr,double ldecr,double pdecr,double pincr,double ptq,double gai,double pastMaxAI,double kl,double aPTQ,double phylPTQ1,double p,string choosePhyllUse)
    {


/*
     Phyllochron Model

    Author: 
    Reference: Modeling development phase in the 
                Wheat Simulation Model SiriusQuality.
                See documentation at http://www1.clermont.inra.fr/siriusquality/?page_id=427
    Instituton: INRA Montpellier
    Abstract: Calculate different types of phyllochron 
    
*/
        double phyllochron;

        phyllochron=0;
        if (choosePhyllUse =="Default")
        {
        	if (leafNumber < ldecr) phyllochron = fixPhyll * pdecr;
        	else if (leafNumber >= ldecr && leafNumber < lincr) phyllochron = fixPhyll;
        	else phyllochron = fixPhyll * pincr;
        }         
        if (choosePhyllUse =="PTQ")
        {
            gai = Math.Max(pastMaxAI,gai);
            pastMaxAI = gai;
            if (gai > 0.0) phyllochron = phylPTQ1 * ((gai * kl) / (1 - Math.Exp(-kl * gai))) / (ptq + aPTQ);
            else phyllochron = phylPTQ1;   
        }        
        if (choosePhyllUse == "Test")
        {
            if (leafNumber < ldecr) phyllochron = p * pdecr;
            else if (leafNumber >= ldecr && leafNumber < lincr)  phyllochron = p;
            else phyllochron = p * pincr;
        }       
        return new Phyllochron(phyllochron,pastMaxAI);
    }
}

File: test/openalea/phyllochron.py
import numpy as np 
from copy import copy
from math import *

def phyllochron(fixPhyll=5.0,
                leafNumber=0.0,
                lincr=8.0,
                ldecr=10.0,
                pdecr=0.4,
                pincr=1.5,
                ptq=0.0,
                gai=0.0,
                pastMaxAI=0.0,
                kl=0.45,
                aPTQ=0.842934,
                phylPTQ1=20.0,
                p=120.0,
                choosePhyllUse='Default'):
    """


    Phyllochron Model
    Author: Pierre Martre
    Reference: Modeling development phase in the 
                Wheat Simulation Model SiriusQuality.
                See documentation at http://www1.clermont.inra.fr/siriusquality/?page_id=427
    Institution: INRA Montpellier
    Abstract: Calculate different types of phyllochron 

    """
    if choosePhyllUse =="Default":
        if (leafNumber < ldecr): phyllochron = fixPhyll * pdecr
        elif (leafNumber >= ldecr and leafNumber < lincr): phyllochron = fixPhyll
        else: phyllochron = fixPhyll * pincr
    if choosePhyllUse =="PTQ":
        pastMaxAI1 = pastMaxAI
        gai = max(pastMaxAI1,gai)
        pastMaxAI = gai
        if (gai > 0.0): phyllochron = phylPTQ1 * ((gai * kl) / (1 - exp(-kl * gai))) / (ptq + aPTQ)
        else: phyllochron = phylPTQ1      
    if choosePhyllUse == "Test":
        if (leafNumber < ldecr): phyllochron = p * pdecr
        elif (leafNumber >= ldecr and leafNumber < lincr): phyllochron = p
        else: phyllochron = p * pincr
    return  phyllochron, pastMaxAI
